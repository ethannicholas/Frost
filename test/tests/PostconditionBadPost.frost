@extendable
class PostconditionBadPost1 {
    var x := 0

    @extendable
    @post(x > 0)
    method test(x:Int) {
        self.x := x
    }
}

class PostconditionBadPost2 : PostconditionBadPost1 {
    @override
    @post(x > 0)
    method test(x:Int) {
        self.x := x
    }

}