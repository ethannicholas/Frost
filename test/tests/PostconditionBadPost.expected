PostconditionBadPost.frost:12:13: error: cannot use '@post' in a method which overrides another method; did you mean '@postAnd'?
1 error
